### ISSUE TITLE
Failed to handle server with uppercase UDN

### ISSUE THREAD

**Issue by dublok** [tkem/mopidy-dleyna#34]
My upnp-server has an uppercase UDN. It is detected by dleyna and stored uppercase, but delivered to mopidy as lowercase (converted by `uricompose()`). The first access to the server throws an exception.

```
INFO     Found digital media server DVBViewer Media Server (MEDIA7) [uuid:37805F2D-361E-4C05-BFBE-03C1951A1866]
...
ERROR    dLeynaBackend backend caused an exception.
Traceback (most recent call last):
  File "/home/jens/mopidy-dev/mopidy/mopidy/core/library.py", line 19, in _backend_error_handling
    yield
  File "/home/jens/mopidy-dev/mopidy/mopidy/core/library.py", line 112, in _browse
    result = backend.library.browse(uri).get()
  File "/usr/lib/python2.7/dist-packages/pykka/threading.py", line 52, in get
    compat.reraise(*self._data['exc_info'])
  File "/usr/lib/python2.7/dist-packages/pykka/compat.py", line 12, in reraise
    exec('raise tp, value, tb')
  File "/usr/lib/python2.7/dist-packages/pykka/actor.py", line 201, in _actor_loop
    response = self._handle_receive(message)
  File "/usr/lib/python2.7/dist-packages/pykka/actor.py", line 295, in _handle_receive
    return callee(*message['args'], **message['kwargs'])
  File "/home/jens/mopidy-dev/mopidy-dleyna/mopidy_dleyna/library.py", line 77, in browse
    return self.__browse_container(uri)
  File "/home/jens/mopidy-dev/mopidy-dleyna/mopidy_dleyna/library.py", line 171, in __browse_container
    future = self.__browse(uri, offset, limit)
  File "/home/jens/mopidy-dev/mopidy-dleyna/mopidy_dleyna/library.py", line 185, in __browse
    future = dleyna.browse(uri, offset, limit, BROWSE_FILTER)
  File "/home/jens/mopidy-dev/mopidy-dleyna/mopidy_dleyna/client.py", line 158, in browse
    baseuri, objpath = self.__parseuri(uri)
  File "/home/jens/mopidy-dev/mopidy-dleyna/mopidy_dleyna/client.py", line 214, in __parseuri
    raise LookupError('Cannot resolve %s' % uri)
LookupError: Cannot resolve dleyna://uuid%3a37805f2d-361e-4c05-bfbe-03c1951a1866
```

I have a workaround which works for me, but I am not sure if it is the best solution:

```
diff --git a/mopidy_dleyna/client.py b/mopidy_dleyna/client.py
index bf25ac3..f3a2cf5 100644
--- a/mopidy_dleyna/client.py
+++ b/mopidy_dleyna/client.py
@@ -84,7 +84,7 @@ class dLeynaServers(collections.Mapping):
             return len(self.__servers)

     def __add_server(self, obj):
-        udn = obj['UDN']
+        udn = obj['UDN'].lower()
         obj['URI'] = uritools.uricompose('dleyna', udn)
         if udn not in self:
             self.__log_server_action('Found', obj)

```

**Comment by tkem** [comment]
Thanks for reporting! Never encountered uppercase UDNs myself, but there are probably lots oft other interoperability issues still in Mopidy-dLeyna...
Will fix ASAP. For the records: What Server/device is that? DVBViewer doesn't ring any bells...

**Comment by tkem** [comment]
This should be fixed in Mopidy-dLeyna v1.0.4 which should be available from PyPi right now.
Please let me know if you still have issues with that, since for lack of a DLNA server with uppercase UDNs this hasn't been tested very thoroughly ;-)
Thanks for reporting and taking the time to track down the cause of this issue in the first place!

**Comment by dublok** [comment]
Your fix is working. Thank you.
DVBViewer is a media center for watching digital TV. It has server application for recording and sharing Sat-IP over network. The server works as UPnP server, too.
